src/wallet/rpcwallet.cpp

```diff
@@ -4567,6 +4567,59 @@ static RPCHelpMan upgradewallet()
     };
 }
 
+RPCHelpMan analyzerawtransaction()
+{
+    return RPCHelpMan{"analyzerawtransaction",
+        "\nCalculate the balance change resulting in the signing and broadcasting of the given transaction.\n" +
+        HELP_REQUIRING_PASSPHRASE,
+        {
+            {"hexstring", RPCArg::Type::STR, RPCArg::Optional::NO, "The transaction hex string"},
+        },
+        RPCResult{
+            RPCResult::Type::NUM, "", "The wallet balance change (negative means decrease)."
+        },
+        RPCExamples{
+            HelpExampleCli("analyzerawtransaction", "\"myhex\"")
+            + HelpExampleRpc("analyzerawtransaction", "\"myhex\"")
+        },
+    [&](const RPCHelpMan& self, const JSONRPCRequest& request) -> UniValue
+{
+    std::shared_ptr<CWallet> const wallet = GetWalletForJSONRPCRequest(request);
+    if (!wallet) return NullUniValue;
+    CWallet* const pwallet = wallet.get();
+
+    RPCTypeCheck(request.params, {UniValue::VSTR}, true);
```

Comment:
```suggestion
    RPCTypeCheck(request.params, {UniValue::VSTR}, /* fAllowNull */ true);
```

Reply:
If you allow `null`, then L4598 will lead to a crash, would it not?

Reply:
The RPCHelpMan framework will not execute the code, since the first param is non-optional.

Reply:
Would `RPCTypeCheck(request.params, {UniValue::VSTR})` work too then?